authority/tls.go

```diff
@@ -426,6 +428,9 @@ func (a *Authority) Revoke(ctx context.Context, revokeOpts *RevokeOptions) error
 
 		// Save as revoked in the Db.
 		err = a.revoke(revokedCert, rci)
+
+		// Generate a new CRL so CRL requesters will always get an up-to-date CRL whenever they request it
+		_, _ = a.GenerateCertificateRevocationList(true)
```

Comment:
I'd like you to change the way of generating CRLs to a background job, and every time a certificate is revoked, a message will be sent to a channel, that channel should probably be waiting on a `select` until a new message is sent or until a timer expires:
```go
select {
case <- a.revokeCh:
	// GenerateCRL
case <-time.After(a.revokeDuration): // it might be good to add some jitter here.
	// GenerateCRL
}

Reply:
With this job, we should rename the method `GenerateCertificateRevocationList` to something like `GetCertificateRevocationList` as it will be always like a static file.